include/seqan3/io/detail/magic_header.hpp

```diff
@@ -0,0 +1,95 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::magic_header.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <array>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+//!\brief Defines a magic byte sequence to disambiguate different compression formats. Default is empty.
+//!\ingroup io
+template <typename header_tag_t>
+inline constexpr std::array<int8_t, 0> magic_header{};
+
+//!\brief A tag signifying a gz compressed file.
+//!\ingroup io
+struct gz_compression
```

Comment:
Is this the best place for the tags?

Reply:
Do you have a better idea? I don‘t care much. But given that we use it right now only in the combination with the magic_header this seems to be a valid first place. If it turns out to be useful in other contexts as well we can move it to some other place.